src/dune_engine/dep.ml

```diff
@@ -311,8 +311,7 @@ module Set = struct
      depending on [(source_tree x)]. Otherwise, we wouldn't clean up stale
      directories in directories that contain no file. *)
   let dir_without_files_dep dir =
-    file_selector
-      (File_selector.create ~dir File_selector.Predicate_with_id.false_)
+    file_selector (File_selector.of_predicate_lang ~dir Predicate_lang.any)
```

Comment:
Should this be `Predicate_lang.empty`?

Reply:
(If yes, how come ho test caught it?)

Reply:
Fixed it. I think it's because we only used it on directories we already knew are empty. So it was just a little perf. optimization. Any glob can be used here.